tensorflow/lite/delegates/gpu/cl/egl_sync.cc

```diff
@@ -21,9 +21,49 @@ namespace tflite {
 namespace gpu {
 namespace cl {
 
+namespace {
+
+PFNEGLCREATESYNCKHRPROC g_eglCreateSyncKHR = nullptr;
+PFNEGLDESTROYSYNCKHRPROC g_eglDestroySyncKHR = nullptr;
+PFNEGLWAITSYNCKHRPROC g_eglWaitSyncKHR = nullptr;
+PFNEGLCLIENTWAITSYNCKHRPROC g_eglClientWaitSyncKHR = nullptr;
+
+bool IsEglSyncSupported(EGLDisplay display) {
+  static bool isAvailable = [display]() -> bool {
+    // EGL_KHR_fence_sync is apparently a display extension
+    const char* extensions = eglQueryString(display, EGL_EXTENSIONS);
+    if (!extensions) {
+      return false;
+    }
+    if (std::strstr(extensions, "EGL_KHR_fence_sync")) {
+      g_eglCreateSyncKHR = reinterpret_cast<PFNEGLCREATESYNCKHRPROC>(
+          eglGetProcAddress("eglCreateSyncKHR"));
+      g_eglDestroySyncKHR = reinterpret_cast<PFNEGLDESTROYSYNCKHRPROC>(
+          eglGetProcAddress("eglDestroySyncKHR"));
+      g_eglWaitSyncKHR = reinterpret_cast<PFNEGLWAITSYNCKHRPROC>(
+          eglGetProcAddress("eglWaitSyncKHR"));
+      g_eglClientWaitSyncKHR = reinterpret_cast<PFNEGLCLIENTWAITSYNCKHRPROC>(
+          eglGetProcAddress("eglClientWaitSyncKHR"));
+    }
+    return g_eglCreateSyncKHR && g_eglDestroySyncKHR && g_eglWaitSyncKHR &&
+           g_eglClientWaitSyncKHR;
+  }();
+  return isAvailable;
+}
+
+#define ENSURE_SYNC_AVALIABLE_ON(display)                           \
```

Comment:
rather than introducing macro which is not encouraged by the style guide, I would just make the above function return absl::Status instead of bool and get rid of this macro.  you can use RETURN_IF_ERROR(YourFunction(display));

Reply:
Thanks, did that!